spmd/tensor/dispatch.py

```diff
@@ -85,8 +88,27 @@ class OutputSharding:
     failed_reason: Optional[str] = None
 
 
+def is_inplace_op(op_call: torch._ops.OpOverload) -> bool:
```

Comment:
does this cover calls such as 
add(a, b, out=a)

or more generally the "out" argument?

Reply:
any op with "out" argument is a separate overload compare to the inplace op. I think the reason is that it might be `add(a, b, out=b)` then it's not technically a inplace op, hence separating. we will need to add "out" argument overload to rules separately i.e. https://github.com/pytorch/pytorch/blob/2f95b61ceaab65580d8dece821693d8b31ecfca8/aten/src/ATen/native/native_functions.yaml#L478

Reply:
actually let me see if it's easy to add `out` variant as well